indico/modules/rb_new/event/controllers.py

```diff
@@ -0,0 +1,95 @@
+# This file is part of Indico.
+# Copyright (C) 2002 - 2019 European Organization for Nuclear Research (CERN).
+#
+# Indico is free software; you can redistribute it and/or
+# modify it under the terms of the GNU General Public License as
+# published by the Free Software Foundation; either version 3 of the
+# License, or (at your option) any later version.
+#
+# Indico is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+# General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with Indico; if not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import unicode_literals
+
+import json
+
+from flask import jsonify
+
+from indico.modules.events.contributions import Contribution
+from indico.modules.events.management.controllers import RHManageEventBase
+from indico.modules.events.sessions.models.blocks import SessionBlock
+from indico.modules.events.sessions.models.sessions import Session
+from indico.modules.rb.controllers.user.event import RHRoomBookingEventBase
+from indico.modules.rb_new.event.forms import BookingListForm
+from indico.modules.rb_new.views.base import WPEventBookingList
+from indico.util.date_time import format_datetime, format_time
+
+
+class RHEventBookingList(RHRoomBookingEventBase):
+    def _process(self):
+        form = BookingListForm(event=self.event)
+        reservations = self.event.reservations
+        has_unlinked_contribs = (Contribution.query.with_parent(self.event)
+                                 .filter(Contribution.is_scheduled,
+                                         Contribution.room_reservation_links == None)  # noqa
+                                 .has_rows())
+        has_unlinked_session_blocks = (SessionBlock.query
+                                       .filter(SessionBlock.session.has(event=self.event),
+                                               SessionBlock.room_reservation_links == None)  # noqa
+                                       .has_rows())
+
+        is_single_day = self.event.start_dt.date() == self.event.end_dt.date()
+        event_rb_params = {'link_type': 'event', 'link_id': self.event.id,
+                           'recurrence': 'single' if is_single_day else 'daily', 'number': 1, 'interval': 'week',
+                           'sd': self.event.start_dt_local.date().isoformat(),
+                           'ed': None if is_single_day else self.event.end_dt_local.date().isoformat(),
+                           'st': format_time(self.event.start_dt_local.time()),
+                           'et': format_time(self.event.end_dt_local.time()),
+                           'text': self.event.room.name if self.event.room else None}
```

Comment:
do we properly filter the None at some point or do we end up searching for `text=None` if the event has no room?

Reply:
`url_for` is filtering them